Backend/TimeTracker.Api/Controllers/ProjectController.cs

```diff
@@ -0,0 +1,181 @@
+        /// <returns>The Id of the newly created tag</returns>
+        [Authorize]
+        [HttpPost("Tag")]
+        public async Task<GenericResponseDTO<int>> AddTag(CreateTagDTO newTag)
+        {
+            var currentUserId = authHelper.GetCurrentUserId(User);
+
+            // Only allow the teacher to tag a project
+            var project = await database.Projects
+                .FirstAsync(x => x.Id == newTag.ProjectId && x.Teacher.Id == currentUserId);
+            
+            if (project == null)
+            {
+                return new GenericResponseDTO<int>()
+                {
+                    Message = "Couldn't find the project",
+                    Success = false
+                };
+            }
+
+            var tag = new Tag()
+            {
+                Name = newTag.Tag,
+                Project = project
+            };
+
+            await database.AddAsync(tag);
+            await database.SaveChangesAsync();
+
+            return new GenericResponseDTO<int>()
+            {
+                Data = tag.Id,
+                Success = true
+            };
+        }
+
+        /// <summary>
+        /// Create a new project
+        /// </summary>
+        /// <param name="newProject">The details of the new project to create</param>
+        /// <returns>The Id of the project that was created</returns>
+        [Authorize]
+        [HttpPost]
+        public async Task<GenericResponseDTO<int>> CreateProject(ProjectCreateDTO newProject)
+        {
+            var currentUser = await authHelper.GetCurrentUser(User, database);
+
+            var response = new GenericResponseDTO<int>() 
+            { 
+                Success = false
+            };
+
+            // Project validation
+            if (string.IsNullOrWhiteSpace(newProject.ClientName))
+            {
+                response.Message = "A project must have a client name.";
+                return response;
+            }
+
+            if (string.IsNullOrWhiteSpace(newProject.ProjectName))
+            {
+                response.Message = "A project must have a project name.";
+                return response;
+            }
+
+            // Create the tags
+            var tags = new List<Tag>();
+            newProject.Tags.ForEach(x => tags.Add(new Tag() 
+            {
+                Name = x
+            }));
+
+            // Create the new project object
+            var project = new Project()
+            {
+                ClientName = newProject.ClientName,
+                Description = newProject.Description,
+                CreatedTime = DateTime.UtcNow,
+                Name = newProject.ProjectName,
+                Teacher = currentUser,
+                InviteCode = Guid.NewGuid().ToString(),
+                Tags = tags
+            };
+
+            await database.AddAsync(project);
+            await database.SaveChangesAsync();
+
+            response.Success = true;
+            response.Data = project.Id;
```

Comment:
Do we want to return the invite code from this endpoint as well?

Reply:
I'm assuming after creating the account the user will be navigated to the main project page which would pull down all the information. Unless you see it working another way or have a different idea for it.

Reply:
Do we want to redirect to a new page that will display some basic project information? We can pull all that info on the next page so this should be good to go.